plugin/pkg/scheduler/cache.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"github.com/golang/groupcache/lru"
+	"hash/adler32"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/algorithm"
+	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
+)
+
+const maxCacheEntries = 4096
+
+type HostPredicate struct {
+	Fit        bool
+	FailReason sets.String
+}
+
+type podCacheEnty struct {
+	Fit        bool
+	FailReason sets.String
+	Score      schedulerapi.HostPriority
+}
+
+type NodeCache struct {
+	predicatesCache *lru.Cache
+	prioritiesCache *lru.Cache
+}
```

Comment:
Should we add a sync.RWMutex for those cache? since the list&watch client of scheduler could invalidate cache(see my another comment),  and scheduler will read the cache when it schedule a pod.